lib/internal/modules/esm/module_job.js

```diff
@@ -75,6 +75,19 @@ const isCommonJSGlobalLikeNotDefinedError = (errorMessage) =>
 const explainCommonJSGlobalLikeNotDefinedError = (e, url) => {
   if (e?.name === 'ReferenceError' &&
       isCommonJSGlobalLikeNotDefinedError(e.message)) {
+
+    if (e.message.includes('require is not defined')) {
+      e.message = [
+        'ERR_AMBIGUOUS_MODULE_SYNTAX: This file cannot be parsed as either CommonJS or ES Module.',
+        '- CommonJS error: await is only valid in async functions.',
+        '- ES Module error: require is not defined in ES module scope.',
+        'If you meant to use CommonJS, wrap top-level await in async function.',
+        'If you meant to use ESM, do not use require().',
```

Comment:
This error message assumes that a top-level await is present, but simply checking for `require is not defined` wouldn’t necessarily imply that, would it?

Reply:
You're absolutely right! My current implementation only checks for the "require is not defined" error but doesn't verify that top-level await is actually present. This could show misleading error messages for files that only use require() without any await.
I should add an additional check to detect top-level await in the source code before showing this specific error message.